azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -44,17 +58,39 @@ func resourceArmSecurityCenterSubscriptionPricing() *schema.Resource {
 					string(security.Standard),
 				}, false),
 			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"AppServices",
+					"ContainerRegistry",
+					"KeyVaults",
+					"KubernetesService",
+					"SqlServers",
+					"SqlServerVirtualMachines",
+					"StorageAccounts",
+					"VirtualMachines",
+				}, false),
+			},
 		},
 	}
 }
 
+func resourceArmSecurityCenterSubscriptionPricingParseId(id string) (string, string, error) {
+	parts := strings.SplitN(id, "/", 7)
+
+	if len(parts) != 7 || parts[2] == "" || parts[6] == "" {
+		return "", "", fmt.Errorf("unexpected format of ID (%s)", id)
+	}
+
+	return parts[2], parts[6], nil
+}
```

Comment:
This should probably be reworked into a parse helper that enables the use of an ID type for this resource, there are many examples of this pattern in the provider.  (For [example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/compute/parse/virtual_machine.go))